gdnative-core/src/class.rs

```diff
@@ -60,71 +69,148 @@ pub trait NativeClassMethods: NativeClass {
     fn register(builder: &crate::init::ClassBuilder<Self>);
 }
 
-/// A reference to a rust native script.
-pub struct NativeRef<T: NativeClass> {
-    this: *mut sys::godot_object,
-    _marker: PhantomData<T>,
+/// A reference to a GodotObject with a rust NativeClass attached.
+#[derive(Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub struct Instance<T: NativeClass> {
+    this: T::Base,
```

Comment:
The name `owner` is more common than `this`,  which might be confusing to people new to Rust and familiar with other OO-languages.

Reply:
I realize that `this` is used a lot in the direct wrapper code. There it references the owner object. Probably this is a convention established by the gdnative headers. I would probably still prefer to *internally* use `owner` rather than `this`.

Reply:
Done!